py/objcell.c

```diff
@@ -30,7 +30,8 @@
 static void cell_print(const mp_print_t *print, mp_obj_t o_in, mp_print_kind_t kind) {
     (void)kind;
     mp_obj_cell_t *o = MP_OBJ_TO_PTR(o_in);
-    mp_printf(print, "<cell %p ", o->obj);
+    MP_STATIC_ASSERT(sizeof(mp_obj_t) == sizeof(mp_uint_t));
```

Comment:
I don't think it makes sense to have this assert here, because this property is already fundamental to the entire runtime.

It's documented somewhat at the top of `py/obj.h`, where `mp_obj_t` is defined (although the comment there could be extended to note that `mp_uint_t` and `mp_int_t` need to be the same size as `mp_obj_t` (see also the definition of `MP_BYTES_PER_OBJ_WORD` in `py/mpconfig.h`).

If anything, this assert could go at the start of `mp_init()`.

Reply:
Did you commit this change?